tiledb/sm/query/ast/test/unit-query-ast.cc

```diff
@@ -0,0 +1,103 @@
+/**
+ * @file unit-query-ast.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2022 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * Tests the `ASTNode`, `ASTNodeVal` and `ASTNodeExpr` classes.
+ */
+
+#include "test/src/helpers.h"
+#include "tiledb/common/common.h"
+#include "tiledb/sm/query/query_ast.h"
+
+#include "catch.hpp"
+
+using namespace tiledb::sm;
+
+TEST_CASE("ASTNode Constructors", "[QueryCondition][ast][constructor]") {
```

Comment:
A general concern about this UT suite: 
I see that all tests rely on the helper method `ast_node_to_str`. This is risky, as this is assumes that the non-trivial recursive method `ast_node_to_str` is actually bug-free and represents the source of truth. Ideally, to be confident about that, the method itself should be unit tested. 
Also please add some tests that cover trees with depth > 2 and mix of operators, applies to both constructor and combine tests

Reply:
A few questions:
1. Where should the tests be located? In their own file?
2. I'm also assuming I'd create trees and write out the representations manually... is this correct?

Reply:
As discussed, we can skip this for now. Resolving.